src/cache.cpp

```diff
@@ -805,6 +805,9 @@ bool convert_header_to_stat(const char* path, const headers_t& meta, struct stat
 
     // size
     pst->st_size = get_size(meta);
+    if(S_ISDIR(pst->st_mode)){
+        pst->st_size = 4096;
+    }
```

Comment:
This should have `} else { pst->st_size = get_size(meta); }` to avoid setting the size twice.